### PR TITLE
#162171038 add role based access control functionality

### PR DESCRIPTION
**Pull request by pitaz**
#**What does this PR do?**
- add role based access control functionality

**Description of Task to be completed?**
- create roles controller
- create adminGuard middleware
- create tests to handle edge cases

**How should this be manually tested?**
- clone the repository `https://github.com/andela/heimdal-ah-backend.git`
- run `npm run install`
- run `npm run start:dev`
- input `api/v1/admin/users` on postman with the `get` method to retrieve all users
- input `api/v1/admin/users/1` on postman with the `put` method to update a user role

**What are the relevant pivotal tracker stories?**
- [#162171038](https://www.pivotaltracker.com/story/show/162171038)

**Screenshots (if appropriate)**
- NA
**Questions:**
- NA

### CODE REVIEW

File: controllers/ArticlesController.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ class ArticlesController {
       tags, body, title, description, image
     } = req.body;
     try {
-      const articleTitle = await ArticleQueryModel.getArticleByTitle();
+      const articleTitle = await ArticleQueryModel.getArticleByTitle(req.body.title);
```
Comment (el-Joft) [inline controllers/ArticlesController.js]: Nice work. I felt you would have destructured the req.body.title

**Review by el-Joft — APPROVED** [review]

**Review by thebere — APPROVED** [review]
